hack/generator/pkg/codegen/pipeline/testdata/TestInjectResourceConversionTestCases/microsoft.person-v20200101_test.golden

```diff
@@ -0,0 +1,349 @@
+	if !match {
+		actualFmt := pretty.Sprint(actual)
+		subjectFmt := pretty.Sprint(subject)
+		result := diff.Diff(subjectFmt, actualFmt)
+		return result
+	}
+
+	return ""
+}
+
+func Test_Person_Spec_WhenSerializedToJson_DeserializesAsEqual(t *testing.T) {
+	parameters := gopter.DefaultTestParameters()
+	parameters.MaxSize = 10
+	properties := gopter.NewProperties(parameters)
+	properties.Property(
+		"Round trip of Person_Spec via JSON returns original",
+		prop.ForAll(RunJSONSerializationTestForPersonSpec, PersonSpecGenerator()))
+	properties.TestingRun(t, gopter.NewFormatedReporter(true, 240, os.Stdout))
+}
+
+// RunJSONSerializationTestForPersonSpec runs a test to see if a specific instance of Person_Spec round trips to JSON and back losslessly
+func RunJSONSerializationTestForPersonSpec(subject Person_Spec) string {
+	// Serialize to JSON
+	bin, err := json.Marshal(subject)
+	if err != nil {
+		return err.Error()
+	}
+
+	// Deserialize back into memory
+	var actual Person_Spec
+	err = json.Unmarshal(bin, &actual)
+	if err != nil {
+		return err.Error()
+	}
+
+	// Check for outcome
+	match := cmp.Equal(subject, actual, cmpopts.EquateEmpty())
+	if !match {
+		actualFmt := pretty.Sprint(actual)
+		subjectFmt := pretty.Sprint(subject)
+		result := diff.Diff(subjectFmt, actualFmt)
+		return result
+	}
+
+	return ""
+}
+
+// Generator of Person_Spec instances for property testing - lazily instantiated by PersonSpecGenerator()
+var personSpecGenerator gopter.Gen
+
+// PersonSpecGenerator returns a generator of Person_Spec instances for property testing.
+func PersonSpecGenerator() gopter.Gen {
+	if personSpecGenerator != nil {
+		return personSpecGenerator
+	}
+
+	generators := make(map[string]gopter.Gen)
+	AddIndependentPropertyGeneratorsForPersonSpec(generators)
+	personSpecGenerator = gen.Struct(reflect.TypeOf(Person_Spec{}), generators)
+
+	return personSpecGenerator
+}
+
+// AddIndependentPropertyGeneratorsForPersonSpec is a factory method for creating gopter generators
+func AddIndependentPropertyGeneratorsForPersonSpec(gens map[string]gopter.Gen) {
+	gens["FamilyName"] = gen.AlphaString()
+	gens["FullName"] = gen.AlphaString()
+	gens["KnownAs"] = gen.AlphaString()
+}
+
+func Test_Person_Status_WhenPropertiesConverted_RoundTripsWithoutLoss(t *testing.T) {
+	parameters := gopter.DefaultTestParameters()
+	parameters.MaxSize = 10
+	properties := gopter.NewProperties(parameters)
+	properties.Property(
+		"Round trip from Person_Status to Person_Status via AssignPropertiesToPersonStatus & AssignPropertiesFromPersonStatus returns original",
+		prop.ForAll(RunPropertyAssignmentTestForPersonStatus, PersonStatusGenerator()))
+	properties.TestingRun(t, gopter.NewFormatedReporter(false, 240, os.Stdout))
+}
+
+// RunPropertyAssignmentTestForPersonStatus tests if a specific instance of Person_Status can be assigned to v20200101storage and back losslessly
+func RunPropertyAssignmentTestForPersonStatus(subject Person_Status) string {
```

Comment:
Still a bit worried about how much test code will get generated for more deeply-nested resources. Is it about 80 lines per type?

Reply:
Not that that's a very useful comment, sorry! I should stop worrying until we see what it's actually like for some real resources.

Reply:
It's a legitimate concern. Something I've got in the back of my mind is to enhande the golden test stuff so we can filter out functions/interfaces that aren't of interest in the specific test.